src/components/attributeTable/AttributeTable.vue

```diff
@@ -41,21 +74,194 @@ export default {
     return {
       headers: [],
       records: [],
+      features: [],
+      selectedRow: [],
       layer: null,
-      source: null,
       loading: true,
       page: 1
     }
   },
   created () {
     this.populateTable()
+
+    if (this.activateTableMapInteraction) {
+      this.activateSelectRowOnMapClick();
+    }
+  },
+  beforeDestroy () {
+    if (this.layer && this.layer.getSource()) {
+      // unregister event after table is closed
+      this.layer.getSource().un('change', this.prepareTableDataAndColumns);
+    }
   },
   watch: {
     layerId () {
       this.populateTable()
+    },
+    features () {
+      this.records = this.features.map(
+        feature => {
+          const record = feature.getProperties();
+          // set feature id
+          record[this.uniqueRecordKeyName] = feature.getId()
```

Comment:
Does feature.getId() always return a valid value? Or can it be undefined / null at some time? Please check https://openlayers.org/en/latest/apidoc/module-ol_Feature-Feature.html#getId especially the part with "remote source".

Reply:
@chrismayer very good observation!
In fact the tested layers do all return valid values from `feature.getId()`. Not only WFS, but also GeoJSON and KML. Apparently all of them are considered as "remote source". 
Getting IDs might be a problem for not-"remote sources" like for layers where features are created manually. However, Currently I do not see a scenario where this happens. We should for sure note this in an issue.

How shall we proceed with this? Fixing or keeping it in mind?

Reply:
Not sure here. Should definitely be tackled but maybe not here. Maybe setting a random ID by setId for non-remote sources would be an option. But this should be done at layer source creation or similar. I created an issue for that #210.